ee/clickhouse/queries/breakdown_props.py

```diff
@@ -67,6 +69,10 @@ def get_breakdown_prop_values(
             INNER JOIN ({get_team_distinct_ids_query(team.pk)}) AS pdi ON e.distinct_id = pdi.distinct_id
             INNER JOIN ({person_subquery}) person ON pdi.person_id = person.id
         """
+    elif column_optimizer.is_using_cohort_propertes:
```

Comment:
So we have "is_using_cohort_properties" which already does a pass to discover if cohorts are used anywhere (including action steps). This addresses your concern of checking actionsteps again!

I just added this bool in all the cases where we decide to join DISTINCT_IDs so that the join will be decided on approrpiately

Reply:
I see, makes sense!

On a slightly unrelated note

https://github.com/PostHog/posthog/blob/master/ee/clickhouse/models/property.py#L162  -> New cohorts query doesn’t respect the `PersonPropertiesMode` type, and always defaults to a subquery.

I'm adding edge cases like these that prop up to the end of the cohort filtering doc